### ISSUE TITLE
Failure building runtime from tag v8.0.3 - missing version of Microsoft.Native.Quic.MsQuic.Schannel

### ISSUE THREAD

**Issue by Wedgetable** [dotnet/runtime#99978]
### Description

I'm trying to build the dotnet/runtime repo from the `v8.0.3` release tag. I'm using PowerShell and running `.\build.cmd libs -c release -a x64`. It always fails with the following error:

> repo\path\src\libraries\System.Net.Quic\src\System.Net.Quic.csproj : error NU1603: Warning As Error: System.Net.Quic depends on Microsoft.Native.Quic.MsQuic.Schannel (>= 2.2.5-ci.444313) but Microsoft.Native.Quic.MsQuic.Schannel 2.2.5-ci.444313 was not found. An approximate best match of Microsoft.Native.Quic.MsQuic.Schannel 2.3.0-ci.449061 was resolved. [repo\path\Build.proj]

I noticed on [nuget.org: Microsoft.Native.Quic.MsQuic.Schannel](https://www.nuget.org/packages/Microsoft.Native.Quic.MsQuic.Schannel/#versions-body-tab) that version 2.2.5-ci.444313 is not available, and even if so, it probably has a security vulnerability like the other pre-2.3.5 versions.

I can build `main` fine, because the NuGet package version used there is 2.3.5 (09db2b5260168545ce3cef669d07cda82f021b9e), which is publicly available.

Happens both on my local developer machine, and on self-hosted Azure agents in a pipeline script.

### Reproduction Steps

```
git clone https://github.com/dotnet/runtime
cd .\runtime
git fetch --tags
git checkout v8.0.3
.\build.cmd libs -c release -a x64
```

### Expected behavior

```
Build succeeded.
    0 Warning(s)
    0 Error(s)
```

### Actual behavior

```
Build FAILED.

repo\path\src\libraries\System.Net.Quic\src\System.Net.Quic.csproj : error NU1603: Warning As Error: System.Net.Quic depends on Microsoft.Native.Quic.MsQuic.Schannel (>= 2.2.5-ci.444313) but Microsoft.Native.Quic.MsQuic.Schannel 2.2
.5-ci.444313 was not found. An approximate best match of Microsoft.Native.Quic.MsQuic.Schannel 2.3.0-ci.449061 was resolved. [repo\path\Build.proj]
    0 Warning(s)
    1 Error(s)
```

### Regression?

This is not an issue in tag `v8.0.2`, because the version of `Microsoft.Native.Quic.MsQuic.Schannel` being used there is 2.2.3, which is still available as a public NuGet package.

### Known Workarounds

I can change the version of this NuGet package in `eng\Versions.props` to
`<MicrosoftNativeQuicMsQuicVersion>2.3.5</MicrosoftNativeQuicMsQuicVersion>`
and the build is successful.

### Configuration

- Which version of .NET is the code running on? **8.0.1**
- What OS and version, and what distro if applicable? **Windows 10 Enterprise, build 19044**
- What is the architecture (x64, x86, ARM, ARM64)? **x64**

**Comment by Wedgetable** [comment]
This seemed to change to a non-public package in commit ddb41ec46ddc558be2f43135a1daa01ebac8e841.

**Comment by ManickaP** [comment]
This is fixed in #99627. I think the tag you're building will stay like that, but release/8.0 branch should work.
